cpp/src/umap/simpl_set_embed/algo.h

```diff
@@ -45,24 +45,24 @@ namespace Algo {
 using namespace ML;
 
 /**
-             * Calculate the squared distance between two vectors of size n
-             */
+ * Calculate the squared distance between two vectors of size n
+ */
 template <typename T>
-__device__ __host__ float rdist(const T *X, const T *Y, int n) {
-  float result = 0.0;
+__device__ __host__ double rdist(const T *X, const T *Y, int n) {
```

Comment:
There is gotta be a ml-prim for this function. Would be nice to use ml-prims whenever possible. I think either you can use distance prims or you can use the following

https://github.com/rapidsai/cuml/blob/branch-0.10/cpp/src_prims/linalg/map_then_reduce.h

One of the prims should cover this.

Reply:
This is a simple function that just happens to be a device function. It's not a kernel.